hrm-domain/packages/hrm-types/Case.ts

```diff
@@ -62,12 +62,18 @@ const getSectionSpecificDataFromNotesOrReferrals = (
 };
 
 export const WELL_KNOWN_CASE_SECTION_NAMES = {
-  households: { getSectionSpecificData: s => s.household, sectionTypeName: 'household' },
+  households: {
+    getSectionSpecificData: (s: any) => s.household,
```

Comment:
I think this could be a `CaseSectionsMap` parameter type?

Reply:
If I do so, the `CaseSectionsMap` type definition errors with 
> Type parameter 'k' has a circular constraint.ts(2313)

I will try to write a proper type but won't spend much here (it was not properly typed before anyways :P).